From 716493abc86ac82144b9bd118aeb9cc61bde9c90 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 7 Jan 2025 12:09:48 +0000
Subject: [PATCH] dts: bcm2712: PL011 UARTs are actually r1p5

The ARM PL011 UART instances in BCM2712 are r1p5 spec, which means they
have 32-entry FIFOs. The correct periphid value for this is 0x00341011.
Thanks to N Buchwitz for pointing this out.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
@@ -618,6 +618,10 @@
 				 IRQ_TYPE_LEVEL_HIGH)>;
 };
 
+&uart10 {
+	arm,primecell-periphid = <0x00341011>;
+};
+
 &aon_intr {
 	status = "disabled";
 };
